src/js/common/components/NavButtons.jsx

```diff
@@ -21,26 +25,42 @@ export default class NavButtons extends React.Component {
       this.props.onSubmit();
     }
   }
-  render() {
-    const { submission, path, pages } = this.props;
-    const currentIndex = pages.indexOf(path);
-    const nextPath = currentIndex + 1 < pages.length ? pages[currentIndex + 1] : null;
-    const previousPath = currentIndex - 1 >= 0 ? pages[currentIndex - 1] : null;
+  goBack() {
+    this.props.onNavigate(this.findNeighbor(-1));
+  }
+  goForward() {
+    this.handleContinue(this.findNeighbor(1));
+  }
+  findNeighbor(increment) {
+    const { pages, path, data } = this.props;
+    const currentIndex = pages.map(page => page.name).indexOf(path);
 
-    const goBack = () => this.props.onNavigate(previousPath);
-    const goForward = () => this.handleContinue(nextPath);
+    for (let i = currentIndex + increment; i >= 0 && i < pages.length; i += increment) {
```

Comment:
I think there are a couple other ways you could do this.

One is to use Lodash's `find`:

```
const find = increment > 0 ? _.find : _.findLast;
const nextPage = find(pages, (page) => {
  return page.depends === undefined || _.matches(page.depends)(data)
}, currentIndex + increment);

return nextPage ? nextPage.name : false;
```
The other would be to filter the pages array in the render method and use the same logic as before to get the previous/next pages. You could also stored the result on the component, but you'd have to make sure you updated it at the right time during the React lifecycle.

```
render() {
  const { submission, path, pages, data } = this.props;
  const filteredPages = pages.filter(page => {
    return page.depends === undefined || _.matches(page.depends)(data)
  });
  const currentIndex = filteredPages.indexOf(path);
  const nextPath = currentIndex + 1 < filteredPages.length ? filteredPages[currentIndex + 1] : null;
  const previousPath = currentIndex - 1 >= 0 ? filteredPages[currentIndex - 1] : null;
  ...
}
```

I like the second option because it's the simplest (unless I'm missing something), but it's also work that happens every `render`, so it's theoretically slower.